netbox/utilities/permissions.py

```diff
@@ -93,6 +96,9 @@ def qs_filter_from_constraints(constraints, tokens=None):
     def _replace_tokens(value, tokens):
         if type(value) is list:
             return list(map(lambda v: tokens.get(v, v), value))
+        User = apps.get_model('users.User')
+        if value == CONSTRAINT_TOKEN_USER and isinstance(tokens.get(CONSTRAINT_TOKEN_USER), User):
+            return tokens[CONSTRAINT_TOKEN_USER].id
```

Comment:
We need to catch the presence of the token inside a list of values as well (above). It might be simpler to just swap out the token value with an integer before we start iterating through the constraints.

Reply:
Agreed, this is much nicer.